src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -94,59 +105,159 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
+      this.enrollErrorMessage += this.generalErrorMessage;
+      this.statusMessageService.showErrorToast('There are some invalid rows');
+      this.isEnrolling = false;
+      return;
+    }
+
+    const enrolledStudents: Student[] = [];
+
+    // Use concat because we cannot afford to parallelize with forkJoin when there's data dependency
+    const enrollRequests: Observable<Students> = concat(
+        ...this.allStudentChunks.map((studentChunk: StudentEnrollRequest[]) => {
+          const studentsEnrollRequest: StudentsEnrollRequest = {
+            studentEnrollRequests: studentChunk,
+          };
+          return this.studentService.enrollStudents(
+              this.courseId, studentsEnrollRequest,
+          );
+        }),
+    );
+
+    enrollRequests.pipe(finalize(() => this.isEnrolling = false)).subscribe({
+      next: (resp: Students) => {
+        enrolledStudents.push(...resp.students);
+      },
+      complete: () => {
+        this.showEnrollResults = true;
+        this.statusMessage.pop(); // removes any existing error status message
+        this.statusMessageService.showSuccessToast('Enrollment successful. Summary given below.');
+
+        const allStudentEntries: StudentEnrollRequest[] = [];
+        this.allStudentChunks.forEach((chunk: StudentEnrollRequest[]) => allStudentEntries.push(...chunk));
+
+        this.enrollResultPanelList =
+            this.populateEnrollResultPanelList(this.existingStudents, enrolledStudents, allStudentEntries);
+
+        this.studentService.getStudentsFromCourse({ courseId: this.courseId }).subscribe((resp: Students) => {
+          this.existingStudents = resp.students;
+          if (!this.isExistingStudentsPanelCollapsed) {
+            const existingStudentTable: Handsontable = this.hotRegisterer.getInstance(this.existingStudentsHOT);
+            this.loadExistingStudentsData(existingStudentTable, this.existingStudents);
+          }
+          this.isExistingStudentsPresent = true;
+        });
+      },
+      error: (resp: ErrorMessageOutput) => {
+        this.enrollErrorMessage = resp.error.message;
+      },
     });
   }
 
+  private checkCellsValidity(section: string, team: string, name: string, email: string, totalStudentCount: number,
+                            index: number, newStudentsHOTInstance: Handsontable,
+                             hotInstanceColHeaders: string[]): boolean {
+    let errorFlag: boolean = false;
+    // Section cannot be null/empty if the total number of students is more than 100
+    if ((section === null || section.trim() === '') && totalStudentCount >= 100) {
+      errorFlag = true;
+      if (!this.enrollErrorMessage.includes(this.sectionErrorMessage)) {
+        this.enrollErrorMessage += this.sectionErrorMessage;
+      }
+    }
+    // Team, name and email cannot be null/empty
+    // Team cannot be in different sections
+    const isTeamRepeated: boolean = !!this.allStudentChunks.find((chunk: StudentEnrollRequest[]) =>
```

Comment:
`this.allStudentChunks` is only populated once `this.numberOfStudentsPerRequest` is hit :) 

This method thus fails to work for errors within every block of 50 students.